libsolidity/interface/StandardCompiler.h

```diff
@@ -83,6 +82,10 @@ class StandardCompiler
 		bool viaIR = false;
 	};
 
+	/// Configures @p _compilerStack according to the input settings in @p _inputsAndSettings.
+	static void configure(InputsAndSettings const& _inputsAndSettings, CompilerStack& _compilerStack);
```

Comment:
Why not make it a factory method? Do you actually need to be able to reconfigure an existing instance of `CompilerStack` in LSP?
```suggestion
	static CompilerStack makeCompilerStack(InputsAndSettings const& _inputsAndSettings, ReadCallback::Callback _readCallback);
```

Also, it might be better to have it as a factory method on `CompilerStack` instead. I think it makes more sense than exposing it as an internal implementation detail of `StandardCompiler`.